main.go

```diff
@@ -42,6 +43,24 @@ func main() {
 		})),
 	)
 	logger := zap.New(core, zap.AddCaller())
+	return logger, traceLevel
+}
+
+func updateRequestID(err error, requestID string) error {
+	if err == nil {
+		return err
+	}
+	usrError, ok := err.(userError.Message)
```

Comment:
The datatype  check are  very good from  programming perspective . should be used minimum if we do not have choice. If we set request id in the provider code , this code is not required

Reply:
Considering the  no. of places the change need to be made , we are fine with the current code